### ISSUE TITLE
istioctl IST0162 wrong report

### ISSUE THREAD

**Issue by zirain** [istio/istio#51257]
1. install istio default profile
2. inject httpbin
3. apply samples/httpbin/httpbin-gateway.yaml
4. run `istioctl analyze`, your will receive following warning
```
Warning [IST0162] (Gateway default/httpbin-gateway) The gateway is listening on a target port (port 80) that is not defined in the Service associated with its workload instances (Pod selector istio=ingressgateway). If you need to access the gateway port through the gateway Service, it will not be available.
```

but actually this's fake report, gateway listen at 8080 and svc export at 80(with target port 8080).

**Comment by leosarra** [comment]
I will take a look

**Comment by leosarra** [comment]
I checked the code of the analyzer and the report seems intentional given what was changed here https://github.com/istio/istio/pull/45726 (bookinfo samples was already updated due to the analyzer failing on it in istio.io https://github.com/istio/istio/pull/45943) 
I'm a bit concerned because if it is indeed intentional then we have to update the samples (httpbin is not the only one) and quite some examples in istio.io

pinging @hanxiaop for his opinion since he was involved in the original changes

**Comment by hanxiaop** [comment]
@leosarra The samples work as expected: if gateway has 80 and service has targetport 8000, it will indeed listen to 8000. The fix made in #45726 may not be sufficient though, which leads to the false positive result.

**Comment by leosarra** [comment]
> The fix made in #45726 may not be sufficient though, which leads to the false positive result.

Then would it make sense to modify the analyzer to verify that the gateway port matches either the target port (if defined) or the port of the service gw?

**Comment by hanxiaop** [comment]
> > The fix made in #45726 may not be sufficient though, which leads to the false positive result.
> 
> Then would it make sense to modify the analyzer to verify that the gateway port matches either the target port (if defined) or the port of the service gw?

I don't think this is quite right, as if the port and targetport are the same, and the gateway has specified a different port, it will still listen to the targetport.
